app/controllers/shops_controller.rb

```diff
@@ -2,4 +2,9 @@ class ShopsController < ApplicationController
   def index
     @shops = Shop.all
   end
+
+  def show
+    @shop = Shop.find(params[:id])
```

Comment:
If you recall the one lesson we had where we wanted to find a shop by its slug? Let's say in theory we have a shop called Julian's Yummy Goods. We would end up wanting a URL like this: `/shops/julian-s-yummy-goods` There are a few steps to this, so I'll use specific code for each part. Within the `ShopsController` we want to be able to do `@shop = Shop.find_by(slug: params[:shop]`. In order to do this, we need to add a slug column to the `shops` table. Once we have that up, we can then create a callback method on the model `before_validation` (Don't remember if it is plural or not) that takes the name of the shop, uses the `pluralize` method to make a human readable URL and set that to the `slug` Below, I'll show you how the route would be setup